### ISSUE TITLE
[Core] Failure in test_sparse_matrices.py

### ISSUE THREAD

**Issue by jrubiogonzalez** [KratosMultiphysics/Kratos#12739]
I'm finding a very strange error inside test_sparse_matrices.test_matrix_assembly. It only fails inside my WSL, on Windows it is ok, in our CI it runs ok. It is only failing in this machine.

`
======================================================================
FAIL: test_matrix_assembly (__main__.TestSparseMatrixInterface)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jrubio/dev/Poesis/kratos/kratos/tests/test_sparse_matrices.py", line 97, in test_matrix_assembly
    self.assertEqual(B.value_data()[i], validation_data[i])
AssertionError: np.float64(2.0) != -3.0

----------------------------------------------------------------------
`

This is the error I get for this modified code to write the outputs:

![image](https://github.com/user-attachments/assets/f512c3fc-5071-48a5-b1df-119666013dca)

And this is the output.
`
This is B:----------------------------------
CsrMatrix
size1 : 3
size2 : 3
nnz : 9
index1_data : 
0,3,6,9,
index2_data : 
0,1,2,0,1,2,0,1,2,
value_data  : 
2,-3,1,-3,6,-3,1,-3,2,

size1 : 3
size2 : 3
nnz : 9
index1_data : 
0,3,6,9,
index2_data : 
0,1,2,0,1,2,0,1,2,
value_data  : 
2,-3,1,-3,6,-3,1,-3,2,

This is B.value_data():----------------------------------
[2. 2. 2. 2. 2. 2. 2. 2. 2.]
%i : 0
[2. 2. 2. 2. 2. 2. 2. 2. 2.]
[2.0, -3.0, 1.0, -3.0, 6.0, -3.0, 1.0, -3.0, 2.0]
[0 0 0 0 0 0 0 0 0]
[0, 1, 2, 0, 1, 2, 0, 1, 2]
%i : 1
[2. 2. 2. 2. 2. 2. 2. 2. 2.]
[2.0, -3.0, 1.0, -3.0, 6.0, -3.0, 1.0, -3.0, 2.0]
[0 0 0 0 0 0 0 0 0]
[0, 1, 2, 0, 1, 2, 0, 1, 2]
F.
`

For some reason when you access through value_data() the output is not right. 

As I said in other machines with different version it works OK, so it looks like some library bug. I'm adding it here for the record just in case somebody finds a similar issue.

**Comment by matekelemen** [comment]
sounds similar to what @roigcarlo has been hunting

**Comment by roigcarlo** [comment]
yep, its suspiciously similar to what we are facing in our CI. I link the PR for reference #12528

**Comment by RiccardoRossi** [comment]
do you have access to the machine? If so, could you give us debug access?? we are very lost at why this happens

**Comment by roigcarlo** [comment]
Also, worth to mention that I faced a similar issue in the past when trying to run tests with mismatching versions of the core and applications (in my partiuclar case was Kratos core + Linear Solvers application). so maybe its worth a check.
